paddle/fluid/pybind/op_function_generator.cc

```diff
@@ -0,0 +1,163 @@
+
+const int BUFFER_SIZE = 4096;
+
+const char* INPUT_TYPE = R"(imperative::NameVarBaseMap)";
+const char* ARGS_TYPE = R"(framework::AttributeMap)";
+
+const char* OUT_VAR_NUM_TYPE = R"(std::map<std::string, size_t>)";
+const char* RETURN_TYPE = R"(imperative::NameVarBaseMap)";
+
+const char* OUTS_INITIALIZER =
+    R"({{"out": VarBase("mul_out")}, "XX": VarBase("mul_xx")}})";
+
+const char* OUT_INITIALIZER_TEMPLATE =
+    R"({"%s", {std::shared_ptr<imperative::VarBase>(new imperative::VarBase(tracer->GenerateUniqueName())) } })";
+
+const char* OP_FUNCTION_TEMPLATE =
+    R"([](const imperative::NameVarBaseMap& ins, const framework::AttributeMap& attrs, 
+    imperative::NameVarBaseMap outs, const std::map<std::string, size_t>& out_nums)
+  {
+    auto tracer = imperative::GetCurrentTracer();
+    if (outs.size() == 0) {
+      if (out_nums.size() == 0) {
+        imperative::NameVarBaseMap outs_ = %s;
+        outs = std::move(outs_);
+      } else {
+        for (auto &pair : out_nums) {
+          for (size_t i = 0; i < pair.second; i ++) {
+            auto var_base_name = tracer->GenerateUniqueName();
+            auto out = new imperative::VarBase(var_base_name);
+            outs[pair.first].emplace_back(std::shared_ptr<imperative::VarBase>(out));
+          }
+        }
+      }
+    }
+    
+    {
+      py::gil_scoped_release release;
+      tracer->TraceOp("%s", std::move(ins), std::move(outs), std::move(attrs));
+      return outs;
+    }
+  }, py::arg("ins"), py::arg("attrs")=framework::AttributeMap(), 
+  py::arg("outs")=imperative::NameVarBaseMap(), 
+  py::arg("out_nums")=std::map<std::string, size_t>())";
+
+const char* PYBIND_TEMPLATE = R"(%s.def("%s", %s);)";
+
+static std::string RefineName(std::string name) {
+  for (auto& e : name) {
+    if (e == '-' || e == '@') {
+      e = '_';
+    }
+  }
+  return name;
+}
+
+static std::vector<std::string> GenerateOpFunctions(
+    const std::string& module_name) {
+  auto& op_info_map = paddle::framework::OpInfoMap::Instance().map();
+
+  std::vector<std::string> op_function_list;
+  for (auto& pair : op_info_map) {
+    auto& op_info = pair.second;
+    auto op_proto = op_info.proto_;
+    if (op_proto == nullptr) {
+      continue;
+    }
+    auto& op_type = op_proto->type();
+
+    // Generate outs initializer
+    std::string outs_initializer = "{";
+
+    for (auto& output : op_proto->outputs()) {
+      auto& out_name = output.name();
+      char out_initializer_buf[BUFFER_SIZE];
+      snprintf(out_initializer_buf, BUFFER_SIZE, OUT_INITIALIZER_TEMPLATE,
+               out_name.c_str(), op_type.c_str(), out_name.c_str());
+      outs_initializer += out_initializer_buf;
+      outs_initializer += ",";
+    }
+    if (outs_initializer.back() == ',') {
+      outs_initializer.pop_back();
+    }
+    outs_initializer += "}";
+
+    // generate op funtcion body
+    char op_function_buf[BUFFER_SIZE];
+    snprintf(op_function_buf, BUFFER_SIZE, OP_FUNCTION_TEMPLATE,
```

Comment:
How about use `paddle::string::Sprintf` instead of fixed max length `BUFFER_SIZE`?

Reply:
Done, thanks.